### PR TITLE
Stop filling `lti_launches` and make a view with the same name for backwards comaptibility

### PR DESCRIPTION
**Pull request by marcospri**
###
Deploying this can be deployed in one go

-  The code deployed will stop using `lti_launches`
-  Later, when the migration is run the now unused table will become a view

### On `main`

- Start with fresh tables

```tox -qe dockercompose -- exec postgres psql -U postgres -c "truncate event,lti_launches cascade"```

- Make a few launches

https://hypothesis.instructure.com/courses/125/assignments/873

- Check the data on lti_launches
```
tox -qe dockercompose -- exec postgres psql -U postgres -c "select * from lti_launches"

 id  |          created           |                context_id                |                  lti_key                   
-----+----------------------------+------------------------------------------+--------------------------------------------
 166 | 2022-08-31 10:17:34.35375  | f3cd019017839c4630358662a05540f2f6ec5f93 | Hypothesisb3be0b33d0b5e4b1cf3aaf04e0e1819a
 167 | 2022-08-31 10:17:59.729614 | f3cd019017839c4630358662a05540f2f6ec5f93 | Hypothesisb3be0b33d0b5e4b1cf3aaf04e0e1819a
(2 rows)
```

- Switch to `events-lti_launches`

- Make more launches (that won't get into lti_launches proper, but "events"
https://hypothesis.instructure.com/courses/125/assignments/873

- No new row in `lti_launches`

```tox -qe dockercompose -- exec postgres psql -U postgres -c "select * from lti_launches"```

- Apply migration 

```hdev alembic upgrade head```

-`lti_launches` is now a view and returns the old and new rows (from events)

 `tox -qe dockercompose -- exec postgres psql -U postgres -c "select * from lti_launches"` 

### Performance

This three queries have similar perf, or at least they all *finish* which is the main aim here.

### New table

```
select 
  DATE_TRUNC('week', event.timestamp) as date, 
  count(*) as lti_launches, 'event'
from event
where type_id = 1
group by DATE_TRUNC('month', event.timestamp)
order by date
```

### Old table
```
select 
  DATE_TRUNC('week', lti_launches.created) as date, 
  count(*) as lti_launches, 'launch'
from lti_launches
group by DATE_TRUNC('month', lti_launches.created)
order by date 
```

### Against subquery

```
select DATE_TRUNC('week', v.created) as date, count(*) as lti_launches
from (

 SELECT
                event.id,
                event.timestamp AS created,
                grouping.lms_id AS context_id,
                application_instances.consumer_key AS lti_key
            FROM event
            JOIN application_instances
                ON event.application_instance_id = application_instances.id
            JOIN grouping
                ON event.course_id = grouping.id
            WHERE event.type_id = 1 ) v

group by DATE_TRUNC('week', v.created)
order by date
```

### CODE REVIEW

File: lms/migrations/versions/9bb2beba95bc_lti_launches_view.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+# revision identifiers, used by Alembic.
+revision = "9bb2beba95bc"
+down_revision = "396f46318023"
+
+
+def upgrade():
+    conn = op.get_bind()
+    launch_event_type_id = conn.execute(
+        "SELECT id from event_type WHERE type = 'configured_launch'"
+    ).fetchone()[0]
```
Comment (jon-betts) [inline lms/migrations/versions/9bb2beba95bc_lti_launches_view.py:22]: This is nice belt and braces in case we change the value

**Review by jon-betts — APPROVED** [review]
I've tried this out with the provided instructions and it works well. I've not run the speed tests, as it's not a good test locally because there isn't enough information, but I would suggest doing that in QA to get a better idea once the update is run there.

**Comment by marcospri** [comment]
> I've tried this out with the provided instructions and it works well. I've not run the speed tests, as it's not a good test locally because there isn't enough information, but I would suggest doing that in QA to get a better idea once the update is run there.

I tested this in meatabase LMS-read. The query that makes up the query seems to behave the same way as the view itself.

File: lms/migrations/versions/9bb2beba95bc_lti_launches_view.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    # Rename the existing table to leave the name free for the view.
+    # We'll delete it after a while.
+    op.execute("ALTER TABLE lti_launches RENAME TO old_lti_launches")
```
[the remark this answers is not in the document]
Comment (marcospri) [reply lms/migrations/versions/9bb2beba95bc_lti_launches_view.py:26]: We are going to get a remainder in every auto-generated migration unfortunately.

**Pull request merged**